src/Analytics/BasicProvider.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Message\Mothership\CMS\Analytics;
+
+/**
+ * @author Samuel Trangmar-Keates
+ *
+ * A basic provider class which is constructed with the reference, params and names
+ * required. Useful for basic providers.
+ *
+ * If there is a more complecated provider, then AnalyticsProviderInterface should
+ * be implemented
+ */
+class BasicProvider implements AnalyticsProvidorInterface
+{
+	/**
+	 * @var string
+	 */
+	private $_name;
+
+	/**
+	 * @var string
+	 */
+	private $_viewRef;
+
+	/**
+	 * @var array
+	 */
+	private $_params;
+
+	/**
+	 * Construct the object with the correct fields.
+	 * 
+	 * @param string $name    The object identifier
+	 * @param string $viewRef The view reference
+	 * @param array  $params  The parameters to pass to the view
+	 */
+	public function __construct($name, $viewRef, $params = [])
```

Comment:
I personally think all of those classes are inflexable and could be replaced with a class like this ^. I created the interface so it would be possible to do it the original way. I think this way is better as the class is more flexible, but I guess it *is* inconsistent. I can add a GoogleAnalytics class, but it leaves this one hanging around unused. Up to you.